src/web/rustdoc.rs

```diff
@@ -1397,6 +1396,20 @@ mod test {
         })
     }
 
+    #[test]
+    // regression test for https://github.com/rust-lang/docs.rs/issues/856
+    fn test_no_trailing_slash() {
+        crate::test::wrapper(|env| {
+            let db = env.db();
+            db.fake_release().name("dummy").version("0.1.0").create()?;
```

Comment:
It'd be good to have a non-default-platform test in here as well.

Reply:
Good idea for the test, this indeed redirects to `/dummy/` for `/target-redirect/x86_64-apple-darwin`.

However the solution is not clear to me. My first idea was to treat the last part of the path as a platform if it was missing a `.`, but then we'll get 404s for `/target-redirect/platform-that-doesnt-exist`. I think the current behavior is better since no one should be redirected here without a trailing slash, they'd have to type the URL in manually.

Reply:
Oh hold on that's a separate issue altogether. I'll add a fix for that too.